Notes_and_Examples/1_protractor_singleVector/1_protractor.xml

```diff
@@ -42,8 +35,15 @@ def vglcfn(e, ans):
           return 'The length of {} is incorrect. Your length: {:.1f}'.format(vec.name, vec.length)
   def check_angle(check, vectors):
       vec = vectors[check['vector']]
+      expected = check['expected']
       tolerance = check.get('tolerance', 2.0)
-      if abs(vec.angle - check['expected']) > tolerance:
+      upper_limit = (expected + tolerance) % 360
+      lower_limit = (expected - tolerance) % 360
+      if upper_limit < lower_limit:  # Can happen when upper_limit=2, lower_limit=-358, for example.
```

Comment:
@e-kolpakov Thanks for the idea. Calculating the angle between the given vector and the unit vector with expected value makes more sense conceptually, even though the resulting code is not much prettier. It is included in an update to https://github.com/open-craft/jsinput-vectordraw/pull/8, specifically in 3d47b982ae06bb8677319e43b5754750184d2b7b; please check it out.